### PR TITLE
fix password encoding

### PR DESCRIPTION
**Pull request by rinick**
fix the encoding of password in dgapi

### CODE REVIEW

File: lib/src/responder/simple/simple_node.dart [hunk 1]
```diff
@@ -713,6 +713,18 @@ class SimpleNode extends LocalNodeImpl {
       int pos = rslt.indexOf('\u0000');
       if (pos >= 0) rslt = rslt.substring(0, pos);
       return rslt;
+    } else if (str.length == 22) {
+      // a workaround for the broken password database, need to be removed later
```
Comment (Pacane) [inline lib/src/responder/simple/simple_node.dart]: wait what? what is causing this?

File: lib/src/responder/simple/simple_node.dart [hunk 2]
```diff
@@ -713,6 +713,18 @@ class SimpleNode extends LocalNodeImpl {
...
+      try{
+        _encryptEngine.reset();
+         _encryptEngine.init(false, _encryptParams);
+         String rslt = UTF8.decode(_encryptEngine.process(Base64.decode(str)));
```
Comment (Pacane) [inline lib/src/responder/simple/simple_node.dart]: rslt -> result?

File: lib/src/responder/simple/simple_node.dart [hunk 3]
```diff
@@ -713,6 +713,18 @@ class SimpleNode extends LocalNodeImpl {
       int pos = rslt.indexOf('\u0000');
       if (pos >= 0) rslt = rslt.substring(0, pos);
       return rslt;
+    } else if (str.length == 22) {
+      // a workaround for the broken password database, need to be removed later
+      try{
+        _encryptEngine.reset();
```
Comment (Pacane) [inline lib/src/responder/simple/simple_node.dart]: no need for underscores

File: lib/src/responder/simple/simple_node.dart [hunk 4]
```diff
@@ -703,7 +703,7 @@ class SimpleNode extends LocalNodeImpl {
     Uint8List utf8bytes = UTF8.encode(str);
     Uint8List block = new Uint8List((utf8bytes.length + 31 )~/32 * 32);
     block.setRange(0, utf8bytes.length, utf8bytes);
-    return Base64.encode(_encryptEngine.process(block));
+    return '\u001Bpw:${Base64.encode(_encryptEngine.process(block))}';
   }
   static String decryptString(String str) {
     if (str.startsWith('\u001Bpw:')) {
```
Comment (Pacane) [inline lib/src/responder/simple/simple_node.dart]: these tokens should be documented, I have no idea what these are about tbh.

File: lib/src/responder/simple/simple_node.dart [hunk 5]
```diff
@@ -713,6 +713,18 @@ class SimpleNode extends LocalNodeImpl {
       int pos = rslt.indexOf('\u0000');
       if (pos >= 0) rslt = rslt.substring(0, pos);
       return rslt;
+    } else if (str.length == 22) {
```
Comment (Pacane) [inline lib/src/responder/simple/simple_node.dart]: Extract `22` to a constant with a clear name showing what it means.

File: lib/src/responder/simple/simple_node.dart (hunk 3 above)
Comment (rinick) [inline lib/src/responder/simple/simple_node.dart]: I need to make it private so the internal engine can't be touched from outside

**Pull request merged**